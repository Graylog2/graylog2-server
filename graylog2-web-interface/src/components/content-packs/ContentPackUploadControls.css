.btn:local(.button) {
    margin-left: 5px;
}
