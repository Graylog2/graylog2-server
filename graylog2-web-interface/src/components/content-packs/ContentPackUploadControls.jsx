import React from 'react';

import UserNotification from 'util/UserNotification';
import { BootstrapModalForm, Input } from 'components/bootstrap';
import { Button } from 'react-bootstrap';
import CombinedProvider from 'injection/CombinedProvider';

import style from './ContentPackUploadControls.css';

const { ContentPacksActions } = CombinedProvider.get('ContentPacks');

class ContentPackUploadControls extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isOpen: false,
    };

    this._openModal = this._openModal.bind(this);
    this._closeModal = this._closeModal.bind(this);
    this._save = this._save.bind(this);
  }

  _openModal() {
    this.setState({ isOpen: true });
    this.uploadModal.open();
  }

  _closeModal() {
    this.uploadModal.close();
  }

  _save(submitEvent) {
    submitEvent.preventDefault();
    if (!this.uploadInput.getInputDOMNode().files || !this.uploadInput.getInputDOMNode().files[0]) {
      return;
    }

    const reader = new FileReader();

    reader.onload = (evt) => {
      const request = evt.target.result;
      ContentPacksActions.create.triggerPromise(request)
        .then(
          () => {
            UserNotification.success('Content pack imported successfully', 'Success!');
            ContentPacksActions.list();
          },
          (response) => {
            const message = 'Error importing content pack, please ensure it is a valid JSON file. Check your ' +
              'Graylog logs for more information.';
            const title = 'Could not import content pack';
            let smallMessage = '';
            if (response.additional && response.additional.body && response.additional.body.message) {
              smallMessage = `<br /><small>${response.additional.body.message}</small>`;
            }
            UserNotification.error(message + smallMessage, title);
          });
    };

    reader.readAsText(this.uploadInput.getInputDOMNode().files[0]);
    this._closeModal();
  }

  render() {
    return (
      <span>
        <Button className={style.button} active={this.state.isOpen} id="upload-content-pack-button" bsStyle="success" onClick={this._openModal}>Upload</Button>
        <BootstrapModalForm
          onModalClose={() => { this.setState({ isOpen: false }); }}
          ref={(node) => { this.uploadModal = node; }}
          onSubmitForm={this._save}
          title="Upload Content Pack"
          submitButtonText="Upload">
          <Input
            ref={(node) => { this.uploadInput = node; }}
            id="upload-content-pack"
            label="Choose File"
            type="file"
            help="Choose Content Pack from disk" />
        </BootstrapModalForm>
      </span>
    );
  }
}

export default ContentPackUploadControls;
