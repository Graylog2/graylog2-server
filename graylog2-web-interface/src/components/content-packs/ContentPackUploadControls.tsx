/*
 * Copyright (C) 2020 Graylog, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the Server Side Public License, version 1,
 * as published by MongoDB, Inc.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * Server Side Public License for more details.
 *
 * You should have received a copy of the Server Side Public License
 * along with this program. If not, see
 * <http://www.mongodb.com/licensing/server-side-public-license>.
 */
import React, { useState, useRef } from 'react';

import UserNotification from 'util/UserNotification';
import { BootstrapModalForm, Input, Button } from 'components/bootstrap';
import { ContentPacksActions } from 'stores/content-packs/ContentPacksStore';
import useProductName from 'brand-customization/useProductName';

import style from './ContentPackUploadControls.css';

const ContentPackUploadControls = () => {
  const productName = useProductName();
  const [isOpen, setIsOpen] = useState(false);
  const uploadInputRef = useRef(null);

  const openModal = () => setIsOpen(true);
  const closeModal = () => setIsOpen(false);

  const save = (submitEvent) => {
    submitEvent.preventDefault();

    const uploadInput = uploadInputRef.current.getInputDOMNode();
    if (!uploadInput.files || !uploadInput.files[0]) {
      return;
    }

    const reader = new FileReader();

    reader.onload = (evt) => {
      const request = evt.target.result;

      ContentPacksActions.create.triggerPromise(request as string).then(
        () => {
          UserNotification.success('Content pack imported successfully', 'Success!');
          ContentPacksActions.list();
        },
        (response) => {
          const message = `Error importing content pack, please ensure it is a valid JSON file. Check your ${productName} server logs for more information.`;
          const title = 'Could not import content pack';
          let smallMessage = '';

          if (response.additional && response.additional.body && response.additional.body.message) {
            smallMessage = `<br /><small>${response.additional.body.message}</small>`;
          }

          UserNotification.error(message + smallMessage, title);
        },
      );
    };

    reader.readAsText(uploadInput.files[0]);
    closeModal();
  };

  return (
    <span>
      <Button
        className={style.button}
        active={isOpen}
        id="upload-content-pack-button"
        bsStyle="info"
        onClick={openModal}>
        Upload
      </Button>
      <BootstrapModalForm
        onCancel={closeModal}
        show={isOpen}
        onSubmitForm={save}
        title="Upload Content Pack"
        submitButtonText="Upload">
        <Input
          ref={uploadInputRef}
          id="upload-content-pack"
          label="Choose File"
          type="file"
          help="Choose Content Pack from disk"
        />
      </BootstrapModalForm>
    </span>
  );
};

export default ContentPackUploadControls;
