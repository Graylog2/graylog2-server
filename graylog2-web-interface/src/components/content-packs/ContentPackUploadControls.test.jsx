import React from 'react';
import renderer from 'react-test-renderer';
import 'helpers/mocking/react-dom_mock';

import ContentPackUploadControls from 'components/content-packs/ContentPackUploadControls';

describe('<ContentPackUploadControls />', () => {
  it('should render', () => {
    const wrapper = renderer.create(<ContentPackUploadControls />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
