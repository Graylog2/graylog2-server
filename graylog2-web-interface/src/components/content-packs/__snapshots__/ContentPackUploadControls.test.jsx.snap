// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentPackUploadControls /> should render 1`] = `
<ContentPackUploadControls>
  <span>
    <ForwardRef
      active={false}
      bsStyle="success"
      className="button"
      id="upload-content-pack-button"
      onClick={[Function]}
    >
      <Button
        active={false}
        bsStyle="success"
        className="button"
        id="upload-content-pack-button"
        onClick={[Function]}
      >
        <StyledComponent
          active={false}
          bsStyle="success"
          className="button"
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "SIZES": Array [
                "large",
                "small",
                "xsmall",
              ],
              "STYLES": Array [
                "success",
                "warning",
                "danger",
                "info",
                "default",
                "primary",
                "link",
              ],
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "Button-c9cbmb-0",
                "isStatic": false,
                "lastClassName": "jKsGyd",
                "rules": Array [
                  [Function],
                ],
              },
              "displayName": "Button",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "Button-c9cbmb-0",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          id="upload-content-pack-button"
          onClick={[Function]}
        >
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="success"
            className="button Button-c9cbmb-0 jKsGyd"
            disabled={false}
            id="upload-content-pack-button"
            onClick={[Function]}
          >
            <button
              className="button Button-c9cbmb-0 jKsGyd btn btn-success"
              disabled={false}
              id="upload-content-pack-button"
              onClick={[Function]}
              type="button"
            >
              Upload
            </button>
          </Button>
        </StyledComponent>
      </Button>
    </ForwardRef>
    <BootstrapModalForm
      cancelButtonText="Cancel"
      formProps={Object {}}
      onCancel={[Function]}
      onModalClose={[Function]}
      onModalOpen={[Function]}
      onSubmitForm={[Function]}
      show={false}
      submitButtonDisabled={false}
      submitButtonText="Upload"
      title="Upload Content Pack"
    >
      <BootstrapModalWrapper
        backdrop="static"
        onClose={[Function]}
        onHide={[Function]}
        onOpen={[Function]}
        showModal={false}
      >
        <Modal
          backdrop="static"
          onHide={[Function]}
          show={false}
        >
          <StyledComponent
            backdrop="static"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "BACKDROP_TRANSITION_DURATION": 150,
                "Body": [Function],
                "Dialog": [Function],
                "Footer": [Function],
                "Header": [Function],
                "SIZES": Array [
                  "large",
                  "small",
                ],
                "TRANSITION_DURATION": 300,
                "Title": [Function],
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Modal-nuree8-0",
                  "isStatic": true,
                  "lastClassName": "cHlGak",
                  "rules": Array [
                    ".modal-content{background-color:",
                    "#FFF",
                    ";border-color:",
                    "rgba(31,31,31,0.2)",
                    ";.modal-header{border-bottom-color:",
                    "#DCE1E5",
                    ";}.modal-footer{border-top-color:",
                    "#DCE1E5",
                    ";}}",
                  ],
                },
                "displayName": "Modal",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Modal-nuree8-0",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onHide={[Function]}
            show={false}
          >
            <Modal
              animation={true}
              autoFocus={true}
              backdrop="static"
              bsClass="modal"
              className="Modal-nuree8-0 cHlGak"
              dialogComponentClass={[Function]}
              enforceFocus={true}
              keyboard={true}
              manager={
                ModalManager {
                  "add": [Function],
                  "containers": Array [],
                  "data": Array [],
                  "handleContainerOverflow": true,
                  "hideSiblingNodes": true,
                  "isTopModal": [Function],
                  "modals": Array [],
                  "remove": [Function],
                }
              }
              onHide={[Function]}
              renderBackdrop={[Function]}
              restoreFocus={true}
              show={false}
            >
              <Modal
                autoFocus={true}
                backdrop="static"
                backdropClassName="modal-backdrop"
                backdropTransition={[Function]}
                containerClassName="modal-open"
                enforceFocus={true}
                keyboard={true}
                manager={
                  ModalManager {
                    "add": [Function],
                    "containers": Array [],
                    "data": Array [],
                    "handleContainerOverflow": true,
                    "hideSiblingNodes": true,
                    "isTopModal": [Function],
                    "modals": Array [],
                    "remove": [Function],
                  }
                }
                onEntering={[Function]}
                onExited={[Function]}
                onHide={[Function]}
                onMouseUp={[Function]}
                renderBackdrop={[Function]}
                restoreFocus={true}
                show={false}
                transition={[Function]}
              />
            </Modal>
          </StyledComponent>
        </Modal>
      </BootstrapModalWrapper>
    </BootstrapModalForm>
  </span>
</ContentPackUploadControls>
`;
